### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by timotheeguerin**
Notable:
- vitest: 2.x
- prettier update that does a minor formatting change by adding parentheses in some ternary expression

### CODE REVIEW

**Review by m-nash — APPROVED** [review]
LGTM for http-client-csharp package

File: packages/typespec-vscode/package.json [hunk 1]
```diff
@@ -25,16 +25,12 @@
...
+    "vscode": "^1.91.0"
   },
   "activationEvents": [
     "onLanguage:typespec",
     "onCommand:typespec.restartServer"
   ],
-  "files": [
-    "typespec-vscode-*.vsix",
-    "ThirdPartyNotices.txt"
-  ],
```
Comment (chrisradek) [inline packages/typespec-vscode/package.json:37]: By removing this, are we including everything now?

Comment (timotheeguerin) [reply packages/typespec-vscode/package.json:37]: yeah the vsce tool complained that we had both files and .vscodeignore.
Vscodeignore used to take priority and files was just for the npm package that contains the vsix, so moved to use npmignore for the package

Comment (timotheeguerin) [reply packages/typespec-vscode/package.json:37]: moved to this pr as this was was just failing for obscure reason https://github.com/microsoft/typespec/pull/3948

**Pull request closed without merging**